model/ChangePermissionsService.php

```diff
@@ -0,0 +1,195 @@
+<?php
+
+namespace oat\taoDacSimple\model;
+
+use core_kernel_classes_Resource;
+use oat\generis\model\OntologyRdfs;
+use oat\tao\model\taskQueue\QueueDispatcherInterface;
+use oat\taoDacSimple\model\tasks\PostChangePermissionsTask;
+
+class ChangePermissionsService
+{
+    private DataBaseAccess $dataBaseAccess;
+    private PermissionsStrategyInterface $strategy;
+    private QueueDispatcherInterface $queueDispatcher;
+
+    public function __construct(
+        DataBaseAccess $dataBaseAccess,
+        PermissionsStrategyInterface $strategy,
+        QueueDispatcherInterface $queueDispatcher
+    ) {
+        $this->dataBaseAccess = $dataBaseAccess;
+        $this->strategy = $strategy;
+        $this->queueDispatcher = $queueDispatcher;
+    }
+
+    public function __invoke(core_kernel_classes_Resource $resource, array $permissionsToSet, bool $isRecursive): void

```

Comment:
To brainstorm later. We generally use __invoke when we do not know the class/method we are calling (like a dynamic class load - middleware in a chain of responsibility, etc) - I am not in this case it is really necessary.